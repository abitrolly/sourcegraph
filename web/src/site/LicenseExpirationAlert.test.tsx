import React from 'react'
import { subMonths, addDays } from 'date-fns'
import { LicenseExpirationAlert } from './LicenseExpirationAlert'
import { mount } from 'enzyme'

describe('LicenseExpirationAlert.test.tsx', () => {
    test('expiring soon', () => {
        expect(mount(<LicenseExpirationAlert expiresAt={addDays(new Date(), 3)} daysLeft={3} />)).toMatchSnapshot()
    })

    test('expired', () => {
        expect(mount(<LicenseExpirationAlert expiresAt={subMonths(new Date(), 3)} daysLeft={0} />)).toMatchSnapshot()
    })
})
