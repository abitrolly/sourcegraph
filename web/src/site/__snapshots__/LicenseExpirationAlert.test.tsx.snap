// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicenseExpirationAlert.test.tsx expired 1`] = `
<LicenseExpirationAlert
  daysLeft={0}
  expiresAt={2005-10-02T22:04:05.000Z}
>
  <DismissibleAlert
    className="alert alert-warning align-items-center "
    partialStorageKey="licenseExpiring.0"
  >
    <div
      className="alert dismissible-alert alert alert-warning align-items-center "
    >
      <div
        className="dismissible-alert__content"
      >
        <Memo(WarningIcon)
          className="icon-inline mr-2 flex-shrink-0"
        >
          <svg
            className="mdi-icon icon-inline mr-2 flex-shrink-0"
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M13,14H11V10H13M13,18H11V16H13M1,21H23L12,2L1,21Z"
            />
          </svg>
        </Memo(WarningIcon)>
        Your Sourcegraph license
         
        expired 3 months ago
        . 
        <AnchorLink
          className="site-alert__link"
          to="/site-admin/license"
        >
          <a
            className="site-alert__link"
            href="/site-admin/license"
          >
            <span
              className="underline"
            >
              Renew now
            </span>
          </a>
        </AnchorLink>
         or 
        <a
          className="site-alert__link"
          href="https://about.sourcegraph.com/contact"
        >
          <span
            className="underline"
          >
            contact Sourcegraph
          </span>
        </a>
      </div>
      <button
        className="btn btn-icon"
        onClick={[Function]}
        type="button"
      >
        <Memo(CloseIcon)
          className="icon-inline"
        >
          <svg
            className="mdi-icon icon-inline"
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </Memo(CloseIcon)>
      </button>
    </div>
  </DismissibleAlert>
</LicenseExpirationAlert>
`;

exports[`LicenseExpirationAlert.test.tsx expiring soon 1`] = `
<LicenseExpirationAlert
  daysLeft={3}
  expiresAt={2006-01-05T22:04:05.000Z}
>
  <DismissibleAlert
    className="alert alert-warning align-items-center "
    partialStorageKey="licenseExpiring.3"
  >
    <div
      className="alert dismissible-alert alert alert-warning align-items-center "
    >
      <div
        className="dismissible-alert__content"
      >
        <Memo(WarningIcon)
          className="icon-inline mr-2 flex-shrink-0"
        >
          <svg
            className="mdi-icon icon-inline mr-2 flex-shrink-0"
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M13,14H11V10H13M13,18H11V16H13M1,21H23L12,2L1,21Z"
            />
          </svg>
        </Memo(WarningIcon)>
        Your Sourcegraph license
         
        will expire in 3 days
        . 
        <AnchorLink
          className="site-alert__link"
          to="/site-admin/license"
        >
          <a
            className="site-alert__link"
            href="/site-admin/license"
          >
            <span
              className="underline"
            >
              Renew now
            </span>
          </a>
        </AnchorLink>
         or 
        <a
          className="site-alert__link"
          href="https://about.sourcegraph.com/contact"
        >
          <span
            className="underline"
          >
            contact Sourcegraph
          </span>
        </a>
      </div>
      <button
        className="btn btn-icon"
        onClick={[Function]}
        type="button"
      >
        <Memo(CloseIcon)
          className="icon-inline"
        >
          <svg
            className="mdi-icon icon-inline"
            fill="currentColor"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"
            />
          </svg>
        </Memo(CloseIcon)>
      </button>
    </div>
  </DismissibleAlert>
</LicenseExpirationAlert>
`;
